core/src/main/java/com/linecorp/armeria/common/FixedHttpRequest.java

```diff
@@ -19,64 +19,191 @@
+        return of(true, headers, content);
+    }
+
+    /**
+     * Creates a new {@link HttpRequest} that will publish the given {@link HttpHeaders} and
+     * {@link HttpObject} with the provided {@code keepAlive}.
+     */
+    public static HttpRequest of(boolean keepAlive, HttpHeaders headers, HttpObject content) {
+        requireNonNull(headers, "headers");
+        requireNonNull(content, "content");
+        return new TwoElementFixedHttpRequest(headers, headers, content, keepAlive);
+    }
+
+    /**
+     * Creates a new {@link HttpRequest} that will publish the given {@code objs}. {@code objs} is not
      * copied so must not be mutated after this method call (it is generally meant to be used with a varargs
-     * invocation). The first element of {@code objs} must be of type {@link HttpHeaders}.
+     * invocation). The first element of {@code objs} must be {@link HttpHeaders}. Keep-alive is enabled.
      */
-    public static FixedHttpRequest of(HttpObject... objs) {
+    public static HttpRequest of(HttpObject... objs) {
         return of(true, objs);
     }
 
     /**
-     * Creates a new {@link FixedHttpRequest} that will publish the given {@code objs}. {@code objs} is not
+     * Creates a new {@link HttpRequest} that will publish the given {@code objs}. {@code objs} is not
      * copied so must not be mutated after this method call (it is generally meant to be used with a varargs
-     * invocation). The first element of {@code objs} must be of type {@link HttpHeaders}.
-     *
-     * @param keepAlive whether to keep the connection alive after this request is handled
+     * invocation). The first element of {@code objs} must be {@link HttpHeaders}.
      */
-    public static FixedHttpRequest of(boolean keepAlive, HttpObject... objs) {
+    public static HttpRequest of(boolean keepalive, HttpObject... objs) {
         requireNonNull(objs, "objs");
-        checkArgument(objs.length > 0, "At least one HttpObject must be published.");
-        checkArgument(objs[0] instanceof HttpHeaders, "First published object must be headers.");
-        return new FixedHttpRequest((HttpHeaders) objs[0], objs, keepAlive);
+        checkArgument(objs.length > 0, "objs must be non-empty");
+        checkArgument(objs[0] instanceof HttpHeaders);
+        HttpHeaders headers = (HttpHeaders) objs[0];
+        switch (objs.length) {
+            case 1:
+                return of(keepalive, headers);
+            case 2:
+                return of(keepalive, headers, objs[0]);
+            default:
+                return new RegularFixedHttpRequest(headers, objs, keepalive);
+        }
     }
 
     /**
-     * Creates a new {@link FixedHttpRequest} that will publish the given {@code objs}. {@code objs} is not
+     * Creates a new {@link HttpRequest} that will publish the given {@code objs}. {@code objs} is not
      * copied so must not be mutated after this method call (it is generally meant to be used with a varargs
```

Comment:
I guess this may be premature optimization. Given this is not true for the case there `objs.length` <= 2, could we just make `RefularFixedHttpRequest` make a copy of `objs`?